rgwadmin/radosgw.go

```diff
@@ -0,0 +1,150 @@
+package rgwadmin
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+const (
+	authRegion = "default"
+	service    = "s3"
+)
+
+var (
+	errNoEndpoint  = errors.New("endpoint not set")
+	errNoAccessKey = errors.New("access key not set")
+	errNoSecretKey = errors.New("secret key not set")
+)
+
+//API struct for New Client
+type API struct {
```

Comment:
Can we different name for this struct, `API` looks more generic. IMO it is good to have something more rgw admin ops specific

Reply:
When you call the package, you might just do `rgwadmin.New()` so I didn't want to duplicate anything. That's why I kept it generic.